%YAML: 1.0
event:
  topic: "/dvs/events"
  freq: 30 # hz
  fx: 287.5352151824009
  fy: 289.4657594005849
  cx: 180.01300190103728
  cy: 145.52597851895223
  resolution: [346, 260]
  deltaT: 0.05 #5e-3 #5e-2 #5e-3 #5e-3 #5e-2 #1.5e-2 #5e-2 #5e-2 #5e-4 #2e-2 #5e-5
  t1: 0.1 #2.5e-3 #2.5e-2 #2.5e-3 #2.5e-2 #1e-2 #4e-2 #5.5e-2 #5.5e-2 #3e-4 #2.5e-2 #3e-5
  mesh_size: 2 # Please keep singular so that there are center pixels present
  timeshift_cam_imu: 0.010493761582439365
  show_events: true #false
  
smooth: 0.1
  
radar_event:
  T_re:              # Radar to -eventcamera external parameters
    - [0.0,  -1.0,  0.0,  0.04077303]
    - [0.0,   0.0,  -1.0,  -0.02330925]
    - [1.0,  0.0,  0.0, -0.16198989]
    - [0, 0, 0, 1]

# Estimator
knot_distance: 0.15 # 0.03     #[s]
update_every_k_knot: 4  #[knot]

radar:
  topic: "/radar/data"
  type: "ars548" #"ti_mmwave"
  Extrinsics:
    time_offset: 0.0
    Trans: [-0.0, 0.0, 0.0]
    Rot: [1, 0, 0,
          0, 1, 0,
          0, 0, 1]
          
event:
  Extrinsics:
    time_offset: 0.0
    # Radar as Main Coordinate
    Trans: [0.04077303, -0.02330925, -0.16198989]
    Rot: [0.0286651,  -0.04739285,  0.99846494,
          0.9992392,   0.0277848,  -0.0273685,
          -0.02644507,  0.99848982,  0.04815325]

output_file: "/home/hao/Desktop/radar-event-new/src/TwistEstimator/output/twist_estimator.csv" 

# twist bias model
bias:
  omega: [1.0, 2.0, 3.0]
  vel: [1.0, 2.0, 3.0]
